### ISSUE TITLE
static compressed index.html is missed when accessing a directory

### ISSUE THREAD

**Issue by glehmann** [joseluisq/static-web-server#178]
**Describe the bug**
The static compressed index.html is missed when accessing a directory.

**To Reproduce**
* create a `doc` directory with an `index.html` file in it
* generate the `index.html.gz` file in the `doc` directory (`gzip -k index.html`) 
* activate the static compression feature
* (optionally) disable the dynamic compression feature
* activate the tracing logs
* run the sws server
* run a GET request on the directory (`xh localhost/doc/`)
* see in the log that SWS tries to access `doc.gz`
* if the dynamic compression was disable, see that the file is serve uncompressed

Note that it uses the statically compressed file when explicitly providing `index.html` in the url (`xh localhost/doc/index.html`).

**Expected behavior**
sws should be able to serve the statically compressed index.html.gz when serving a directory.

**Logs/Screenshots**

```
2023-02-28T06:46:56.326898Z  INFO static_web_server::handler: incoming request: method=GET uri=/doc/
2023-02-28T06:46:56.326912Z TRACE static_web_server::static_files: dir: base="/public", route="doc/"
2023-02-28T06:46:56.326928Z TRACE static_web_server::compression_static: preparing pre-compressed file path variant of /public/doc
2023-02-28T06:46:56.326947Z TRACE static_web_server::compression_static: getting metadata for pre-compressed file variant /public/doc.gz
2023-02-28T06:46:56.326984Z DEBUG static_web_server::static_files: file not found: "/public/doc.gz" Os { code: 2, kind: NotFound, message: "No such file or directory" }
2023-02-28T06:46:56.327003Z TRACE static_web_server::static_files: getting metadata for file /public/doc
2023-02-28T06:46:56.327018Z TRACE static_web_server::static_files: file found: "/public/doc"
2023-02-28T06:46:56.327029Z DEBUG static_web_server::static_files: dir: appending an index.html to the directory path
2023-02-28T06:46:56.327048Z TRACE static_web_server::static_files: file found: "/public/doc/index.html"
```

**Environment and Specs**
 - static-web-server: 2.14.2
 - OS: k8s on linux
 - Client: xh 0.16.1, chrome 110.0.5481.77

**Comment by joseluisq** [comment]
Can you please share also how are you calling SWS like options used or config settings?

**Comment by glehmann** [comment]
sure, it's run with `static-web-server --config-file /sws.toml` and `sws.toml` is 

```toml
[general]
root = "/public"
log-level = "trace"
# we want to serve the pre-compressed files, but don't want to compress them on the fly
compression = false
compression-static = true
```

**Comment by joseluisq** [comment]
I just checked this and effectively, SWS is not appending an `index.html` for directories therefore the issue.
A bugfix should come in next hours.

**Comment by joseluisq** [comment]
Fixed by #186.
Thanks for reporting it 👍
